### ISSUE TITLE
Document how Razor Pages handles HEAD requests

### ISSUE THREAD

**Issue by guardrex** [aspnet/Docs#6124]
> Razor Pages will now fall back to calling a matching GET page handler if no HEAD page handler is defined.

**Comment by guardrex** [comment]
@pranavkm I'm going to have to be on-hold with this one until ur free to look.

I can't get the `OnGet` to handle HEAD requests unless I explicitly do a ...

```csharp
.AddRazorPagesOptions(options =>
{
    options.AllowMappingHeadRequestsToGetHandler = true;
});
```
... somehow that's defaulting to `false` here.

SDK: 2.1.300-rc1-008673
TFM: `netcoreapp2.1`
Microsoft.AspNetCore.App: 2.1.0-rc1-30682
Self-contained test app: `win10-x64`

**Comment by pranavkm** [comment]
It needs the 2.1 compat switch: `services.AddMvc().SetCompatibilityVersion(Microsoft.AspNetCore.Mvc.CompatibilityVersion.Version_2_1);` which effectively sets `AllowMappingHeadRequestsToGetHandler = true`

**Comment by guardrex** [comment]
... ok ... a bit confusing tho since the TFM and "App" package imply that the app is compatible with 2.1 (RC).

Will the need for setting `SetCompatibilityVersion` go away with 2.1 RTM? ... thus I can document this as advertised? ...

> Razor Pages will now fall back to calling a matching GET page handler if no HEAD page handler is defined.

**Comment by pranavkm** [comment]
> Will the need for setting SetCompatibilityVersion go away with 2.1 RTM? ... thus I can document this as advertised? ...

No, it's how you opt in to behavior that is considered breaking between minor releases. By default, existing apps that upgrade to 2.1 do not get this feature; you have to opt in to it.

**Comment by guardrex** [comment]
I see it now ...

> 5. In the ConfigureServices method, change the call to add MVC services to set the compatibility version to 2.1 like so: services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_1);

@Rick-Anderson I suspect this is going to be a problem for scenarios that use these features that rely on the setting but don't make mention of it. It probably isn't enough to merely cover the new bits and assume they picked up on the coverage in the *App startup* topic. 😖 *Why isn't this working?!* 😖

Shall we always link https://docs.microsoft.com/aspnet/core/fundamentals/startup#setcompatibilityversion-for-aspnet-core-mvc (`xref:fundamentals/startup#setcompatibilityversion-for-aspnet-core-mvc`) to new 2.1 scenarios?

**Comment by Rick-Anderson** [comment]
@guardrex  asks: 

>Will the need for setting `SetCompatibilityVersion` go away with 2.1 RTM? ...

It goes away with 3.0 preview 1 and higher.

**Comment by guardrex** [comment]
I think I can work it in because there's (I guess) usually an option that will activate the 2.1 scenario even when compat isn't requested. In this case, the setting is `AllowMappingHeadRequestsToGetHandler`. That presents an opportunity to mention the compat setting for the patch release and link to it. I almost have this one done, so we can see how it looks "on paper" and decide if this approach works.
